assets/js/components/KeyMetrics/MetricsSelectionPanel/Footer.js

```diff
@@ -144,47 +162,42 @@ export default function Footer( { savedMetrics } ) {
 		} );
 
 		if ( ! error ) {
-			setValue( KEY_METRICS_SELECTION_PANEL_OPENED_KEY, false );
 			trackEvent( trackingCategory, 'metrics_sidebar_save' );
+
 			if ( keyMetricsEnabled && isGA4Connected ) {
 				if ( hasMissingCustomDimensions ) {
 					setValues( FORM_CUSTOM_DIMENSIONS_CREATE, {
 						autoSubmit: true,
 					} );
 
-					if ( ! hasAnalytics4EditScope ) {
-						setPermissionScopeError( {
-							code: ERROR_CODE_MISSING_REQUIRED_SCOPE,
-							message: __(
-								'Additional permissions are required to create new Analytics custom dimensions.',
-								'google-site-kit'
-							),
-							data: {
-								status: 403,
-								scopes: [ ANALYTICS_EDIT_SCOPE ],
-								skipModal: true,
-							},
-						} );
+					if ( ! hasAnalytics4EditScope && OAuthURL ) {
+						snapshotStore( CORE_FORMS, registry );
+						navigateTo( OAuthURL );
```

Comment:
Maybe we can introduce new property in the `setPermissionScopeError` object, like `snapshotStore` that would accept store name, in this case `CORE_FORMS`, and then let `AuthenticatedPermissionsModal` handle this in centralised manner, by [checking](https://github.com/google/site-kit-wp/blob/ddbc12a5f33a76d6feaae8451d72a91d958e89ac/assets/js/components/PermissionsModal/AuthenticatedPermissionsModal.js#L62)  if `permissionsError?.snapshotStore`, to snapshot only passed custom store name, otherwise fallback to default snapshot of all stores that implement that method, like it is doing now.

This way it would be less "hacky" and no need to include all these dependencies like `registry`, and `snapshotStore` just for this, and it will be easier to use it in the future by sticking to the common pattern. It also requires less code.

Reply:
Updated, thank you!